xarray/core/dataset.py

```diff
@@ -6368,5 +6368,65 @@ def idxmax(
             )
         )
 
+    def argmin(self, dim=None, axis=None, **kwargs):
+        """Apply argmin() to each variable in the Dataset
+        """
+        if dim is None and axis is None:
+            warnings.warn(
+                "Once the behaviour of DataArray.argmin() and Variable.argmin() with "
+                "neither dim nor axis argument changes to return a dict of indices of "
+                "each dimension, for consistency it will be an error to call "
+                "Dataset.argmin() with no argument, since we don't return a dict of "
+                "Datasets.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+        if (
+            dim is None
+            or axis is not None
+            or not isinstance(dim, Sequence)
+            or isinstance(dim, str)
+        ):
+            # Return int index if single dimension is passed, and is not part of a
+            # sequence
+            argmin_func = getattr(duck_array_ops, "argmin")
```

Comment:
we'll need to preserve `Dataset.attrs` if `keep_attrs=True`

Reply:
I think this is taken care of by `self.reduce()`?

Reply:
yes you are right. sorry for the noise.